### PR TITLE
test: refactor `StructCoder` tests

### PR DESCRIPTION
**Pull request by danielbate**
Related to #753

This PR will:
- Introduce a seperate test file for the service under test
- Seperate test cases for `encode` and `decode`
- Remove snapshot tests to bring expectations into the same test file
- Remove redundant tests to more clearly report on code coverage

### CODE REVIEW

File: packages/abi-coder/src/coders/struct.test.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    ).toThrow('Invalid struct TestStruct');
+  });
+
+  it('should throw type error with missing input for second coder', () => {
+    expect(() =>
+      coder.encode(
+        // @ts-expect-error
+        { a: true }
+      )
+    ).toThrow('Invalid struct TestStruct');
```
Comment (arboleya) [inline packages/abi-coder/src/coders/struct.test.ts:61]:
```suggestion
    ).toThrow(`Invalid struct ${STRUCT_NAME}`);
```

File: packages/abi-coder/src/coders/struct.test.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  });
+
+  it('should throw type error with missing input for first coder', () => {
+    expect(() =>
+      coder.encode(
+        // @ts-expect-error
+        { b: bn(1234) }
+      )
+    ).toThrow('Invalid struct TestStruct');
```
Comment (arboleya) [inline packages/abi-coder/src/coders/struct.test.ts:70]:
```suggestion
    ).toThrow(`Invalid struct ${STRUCT_NAME}`);
```

File: packages/abi-coder/src/coders/struct.test.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  });
+
+  it('should throw type error with invalid input for first coder and missing input for second', () => {
+    expect(() =>
+      coder.encode(
+        // @ts-expect-error
+        { a: 1234 }
+      )
+    ).toThrow('Invalid struct TestStruct');
```
Comment (arboleya) [inline packages/abi-coder/src/coders/struct.test.ts:79]:
```suggestion
    ).toThrow(`Invalid struct ${STRUCT_NAME}`);
```

File: packages/abi-coder/src/coders/struct.test.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  });
+
+  it('should throw type error with invalid input for second coder and missing input for first', () => {
+    expect(() =>
+      coder.encode(
+        // @ts-expect-error
+        { b: true }
+      )
+    ).toThrow('Invalid struct TestStruct');
```
Comment (arboleya) [inline packages/abi-coder/src/coders/struct.test.ts:88]:
```suggestion
    ).toThrow(`Invalid struct ${STRUCT_NAME}`);
```

File: packages/abi-coder/src/coders/struct.test.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  });
+
+  it('should throw type error with invalid input key', () => {
+    expect(() =>
+      coder.encode(
+        // @ts-expect-error
+        { nope: 1234, alsoNope: true }
+      )
+    ).toThrow('Invalid struct TestStruct');
```
Comment (arboleya) [inline packages/abi-coder/src/coders/struct.test.ts:97]:
```suggestion
    ).toThrow(`Invalid struct ${STRUCT_NAME}`);
```

**Review by Torres-ssf — APPROVED** [review]
Good stuff :+1:

**Pull request merged**